js/api/BackgroundProcess.js

```diff
@@ -11,55 +13,156 @@ export const BackgroundGeolocation = Platform.OS === 'ios' ?
+      config = {
+        // License validations
+        orderId: Settings.licenses.BackgroundGeolocation.bundleId,
+        license: Settings.licenses.BackgroundGeolocation.key,
+
+        // Geolocation config
+        desiredAccuracy: 100,
+        distanceFilter: 80,
+        locationUpdateInterval: 30000,
+        fastestLocationUpdateInterval: 30000,
+
+        // Activity Recognition config
+        minimumActivityRecognitionConfidence: 80,
+        activityRecognitionInterval: 45000,
+        stopDetectionDelay: 0,
+        stopTimeout: 2,
+
+        // Application config
+        debug: false,
+
+        // Android config
+        forceReloadOnLocationChange: false,
+        forceReloadOnMotionChange: false,
+        forceReloadOnGeofence: false,
+        stopOnTerminate: true,
+        startOnBoot: true,
+
+        // iOS config
+        stationaryRadius: 200, // restart tracking after the user has moved 200m
+        useSignificantChangesOnly: true,
+        disableMotionActivityUpdates: true,
+      };
+    } else {
+      // High Accuracy Mode
+      console.log('Configuring Geolocation: High Accuracy Mode');
+      config = {
+        // License validations
+        orderId: Settings.licenses.BackgroundGeolocation.bundleId,
+        license: Settings.licenses.BackgroundGeolocation.key,
+
+        // Geolocation config
+        desiredAccuracy: 10,
+        distanceFilter: 50,
+        locationUpdateInterval: 10000,
+        fastestLocationUpdateInterval: 10000,
+
+        // Activity Recognition config
+        minimumActivityRecognitionConfidence: 80,
+        activityRecognitionInterval: 15000,
+        stopDetectionDelay: 1,
+        stopTimeout: 5,
+
+        // Application config
+        debug: false,
+
+        // Android config
+        forceReloadOnLocationChange: false,
+        forceReloadOnMotionChange: false,
+        forceReloadOnGeofence: false,
+        stopOnTerminate: true,
+        startOnBoot: true,
+
+        // iOS config
+        stationaryRadius: 100, // restart tracking after the user has moved 100m
+        useSignificantChangesOnly: false,
+        disableMotionActivityUpdates: false,
+      }
+    }
+    if (options.isInitial) {
+      BackgroundGeolocation.configure(config, callback);
+    } else {
+      BackgroundGeolocation.setConfig(config);
+      if (callback) callback();
+    }
   } catch (e) {
     console.error(e)
   }
 }
 
+/**
+ * Stops any running background services and initializes a new process.
+ */
 export function initializeGeolocationService() {
-  configureGeolocationService((state) => {
+  BackgroundGeolocation.stop();
+  configureGeolocationService({isInitial: true}, (state) => {
+
     BackgroundGeolocation.on('error', function(error) {
-      // printTimelog('error')
+      printTimelog('error');
       console.log(error.type + " Error: " + error.code)
-    })
+    });
 
     // Create initial geofence hooks.
-    setupGeofences()
+    setupGeofences(()=>{
```

Comment:
I had several issues on the emulator and real devices if the geofences were added before the plugin 'starts'.  In fact, one would think the plugin is broken as the geofences would never trigger. Please see: https://github.com/ecohealthalliance/GoodQuestion/blob/geolocation-tests/App.js#L68-L77.